shacl12-inf-rules/index.html

```diff
@@ -160,444 +223,1880 @@
+
+        <p>TODO</p>
+
+        <p class="ednote">RFC 2119 language should autmatically be inserted here.</p>
+        
+      </section>
+    </section>
+
+    <section id="overview">
+      <h2>Outline</h2>
+
+      <p>SHACL rules infer new triples. The input is a data graph and a shape
+      graph with rules, the output is a graph of inferred triples that do not
+      occur in the data graph.</p>
+
+      <p>Data:</p>
+<pre>
+   :A :fatherOf :X .
+   :B :motherOf :X .
+   :C :motherOf :A .
+</pre>
+     <p>Rules:</p>
+<pre>
+RULE { ?x :childOf ?y } WHERE { ?y :fatherOf ?x }
+RULE { ?x :childOf ?y } WHERE { ?y :motherOf ?x }
+
+RULE { ?x :descendedFrom ?y } WHERE { ?x :childOf ?y }
+RULE { ?x :descendedFrom ?y } WHERE { ?x :childOf ?z . ?z :childOf ?y }
+</pre>
+
+      <p>will conclude that: :X is the child of :a and :B. and that :X is descendedFrom :C<p>
+    </section>
+
+    <section id="rules-abstract-syntax">
+        <!-- Export -->
+        <h2>Shape Rules Abstract Syntax</h2>
+
+        <p>The Shape Rules Abstract Syntax</p>
+
+        <p class="ednote">?? Abstract Rules Syntax</p>
+
+        <dl>
+          <dt><dfn data-lt="variable">variable</dfn></dt>
+          <dd><code>[ sh:var "name" ]</code>
+          </dd>
+
+          <dt><dfn data-lt="expression">node expression</dfn></dt>
+          <dd>
+            <em>Link to shacl12-node-expr#?</em> -- needs to be single value, list arguments.
+            Often called an <em>expression</em>.
+          </dd>
+
+          <dt><dfn data-lt="data-block">data block</dfn></dt>
+          <dd>
+            A <em>data block</em> is a set of triples.
+            These form extra facts that are included in the inference process.
+          </dd>
+          
+          <dt><dfn>triple template</dfn></dt>
+          <dd>
+            A <em>triple template</em> is 3-tuple where each element is either
+            a variable or an RDF term (including being a triple term).
+            [=Triple templates=] appear in the [=head=] of a [=rule=].
+          </dd>
+
+          <dt><dfn>triple pattern</dfn></dt>
+          <dd>
+            A <em>triple pattern</em> is 3-tuple where each element is either a
+            variable, or an RDF term  (including being a triple term).
+            [=Triple  patterns=] appear in the [=body=] of a [=rule=]. 
+          </dd>
+
+          <dt><dfn>condition expression</dfn></dt>
+          <dd>
+            A <em>condition expression</em> is a function, or functional form,
+            that evaluates to true or false.
+            [=Condition expressions=] appear in the [=body=] of a [=rule=].
+          </dd>
+
+          <dt><dfn>assignment</dfn></dt>
+          <dd>
+            An <em>assignment</em> is a pair of a <a>variable</a>, called the
+            <dfn>assignment variable</dfn>, and an <a>expression</a>, called the
+            <dfn>assignment expression</dfn>.
+            [=Assignments=] appear in the [=body=] of a [=rule=].
```

Comment:
```suggestion
            An <em>assignment</em> is a pair of a <a>variable</a> and an 
            <a>expression</a>, called the <dfn>assignment variable</dfn>
            and <dfn>assignment expression</dfn>, respectively.
            [=Assignments=] appear in the [=body=] of a [=rule=].
```

Reply:
I prefer the original as it more closely connects the part of the assignment wth its <dfn>.